internal/middleware/wrappedstream.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+// Copyright 2016 Michal Witkowski. All Rights Reserved.
```

Comment:
Note that I'm keeping the copyright notice from the original sources.

Reply:
Can you merge this PR as is, so that I can move with it elsewhere? I promise to send a one-liner in a PR to change this soon.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/4521 to track this